src/Symfony/Component/Translation/Dumper/FileDumper.php

```diff
@@ -114,7 +114,10 @@ protected function formatCatalogue(MessageCatalogue $messages, $domain, array $o
      *
      * @deprecated since version 2.8, to be removed in 3.0. Overwrite formatCatalogue() instead.
      */
-    abstract protected function format(MessageCatalogue $messages, $domain);
+    protected function format(MessageCatalogue $messages, $domain)
+    {
+        throw new \RuntimeException('The '.__METHOD__.' method needs to be implemented to use it.');
```

Comment:
a LogicException would be better (consistent with the way the Console component does such thing).

and the message is bad:

- it should not use ``__METHOD__`` as the method should be implemented in the child class, while it shows ``FileDumper::format`` in your case
- the message is not accurate. It should tell to implement either format or formatCatalogue.